src/streaming/array.dimensions.hh

```diff
@@ -56,6 +58,40 @@ class ArrayDimensions
     const ZarrDimension& height_dim() const;
     const ZarrDimension& width_dim() const;
 
+    /**
+     * @brief Get the dimension at the given index in storage dimension order.
+     * @param idx The index in storage order.
+     * @return The dimension at the given index.
+     */
+    const ZarrDimension& storage_dimension(size_t idx) const;
+
+    /**
+     * @brief Check if dimensions need transposition
+     * @return True if dimensions are not in storage dimension order.
+     */
+    bool needs_transposition() const;
+
+    /**
+     * @brief Check if spatial dimensions (Y, X) need transposition.
+     * @return True if the last two dimensions are swapped between acquisition
+     *         and storage order.
+     */
+    bool needs_spatial_transposition() const;
```

Comment:
This name could be clearer since there are potentially other spatial dimensions.

```suggestion
    bool needs_xy_transposition() const;
```